inc/Modules/Rest/Basic_Options_Controller.php

```diff
@@ -0,0 +1,609 @@
+
+			// Brand site not connected.
+			if ( empty( $api_key ) ) {
+				$failed_sites[] = [
+					'site_name' => self::get_sitename_by_url( $site_url ),
+					'url'       => $site_url,
+					'message'   => __( 'API key not found', 'onemedia' ),
+				];
+				continue;
+			}
+
+			// Perform health check request.
+			$response = wp_safe_remote_get(
+				$site_url . '/wp-json/' . Abstract_REST_Controller::NAMESPACE . '/health-check',
+				[
+					'timeout' => self::HEALTH_CHECK_REQUEST_TIMEOUT, // phpcs:ignore WordPressVIPMinimum.Performance.RemoteRequestTimeout.timeout_timeout
+					'headers' => [
+						'X-OneMedia-Token' => $api_key,
+						'Cache-Control'    => 'no-cache, no-store, must-revalidate',
+					],
+				]
+			);
+
+			if ( is_wp_error( $response ) ) {
+				$failed_sites[] = [
+					'site_name' => self::get_sitename_by_url( $site_url ),
+					'url'       => $site_url,
+					'message'   => $response->get_error_message(),
+				];
+				continue;
+			}
+
+			$response_code = wp_remote_retrieve_response_code( $response );
+			if ( 200 === $response_code ) {
+				continue;
+			}
+
+			$failed_sites[] = [
+				'site_name' => self::get_sitename_by_url( $site_url ),
+				'url'       => $site_url,
+				'message'   => sprintf(
+				/* translators: %d is the HTTP response code. */
+					__( 'HTTP %d response', 'onemedia' ),
+					$response_code
+				),
+			];
+		}
+
+		if ( ! empty( $failed_sites ) ) {
+			$failed_sites_list = [];
+			foreach ( $failed_sites as $failed_site ) {
+				$site_name = $failed_site['site_name'];
+				if ( in_array( $site_name, $failed_sites_list, true ) ) {
+					continue;
+				}
+
+				$failed_sites_list[] = $site_name;
+			}
+
+			return [
+				'success'      => false,
+				'failed_sites' => $failed_sites,
+				'message'      => sprintf(
+				/* translators: %s is the list of unreachable sites. */
+					__( 'Please check your connection for unreachable sites: %s.', 'onemedia' ),
+					implode( ', ', $failed_sites_list )
+				),
+			];
+		}
+
+		return [
+			'success'      => true,
+			'failed_sites' => $failed_sites,
+			'message'      => __( 'All connected sites are reachable.', 'onemedia' ),
+		];
+	}
+
+	/**
+	 * Get OneMedia sync site URLs postmeta value.
+	 *
+	 * @param int $attachment_id The attachment ID.
+	 *
+	 * @return array The array of sync site URLs.
+	 */
+	private static function get_sync_site_urls_postmeta( int $attachment_id ): array {
+		$sites = self::get_sync_sites_postmeta( $attachment_id );
+		if ( empty( $sites ) ) {
+			return [];
+		}
+
+		$site_urls = [];
+		foreach ( $sites as $site ) {
+			if ( ! isset( $site['site'] ) ) {
+				continue;
+			}
+
+			$site_urls[] = untrailingslashit( esc_url_raw( $site['site'] ) );
+		}
+		return $site_urls;
+	}
+
+	/**
+	 * Get OneMedia sync sites postmeta value.
+	 *
+	 * @param int $attachment_id The attachment ID.
+	 *
+	 * @return array The array of sync sites.
+	 */
+	public static function get_sync_sites_postmeta( int $attachment_id ): array {
```

Comment:
Generally, please try and order order classes by `public static` > `public` > `protected`, `private`, `protected static` > `private static`. Makes it clear to see what's actually contributing to the api surface without diving deep into every class file.

2. As mentioned elsewhere there shouldn't be _any_ `public static` methods in our REST controllers. (At most, it'd be an accessor _to_ the REST endpoint, or cross-logic _solely_ between other REST controllers)